test/system/331-system-check.bats

```diff
@@ -0,0 +1,202 @@
+#!/usr/bin/env bats   -*- bats -*-
+#
+# Creates errors that should be caught by `system check`, and verifies
+# that they are caught and remedied, even if it requires discarding some
+# data in read-write layers.
+#
+
+load helpers
+
```

Comment:
This module urgently needs a `teardown()`: otherwise, any individual test failure will leave the system in an undefined and probably unrecoverable state. I don't know what said `teardown()` will look like; it's possible that it will involve `system reset -f`.

Or, perhaps better, run all these tests with temporary storage; see https://github.com/containers/podman/blob/main/test/system/330-corrupt-images.bats

Reply:
I'm not sure that the current testing framework makes it possible to do any of that for a subset of the tests, when they are being run as part of a remote testing suite.

Reply:
True: there is no way to use `--root` et al with `podman-remote`. Is there any actual reason to run these tests under `podman-remote`?

Reply:
I expected that we would be keenly interested in having this function available for Podman Desktop and podman-machine cases.

Reply:
I know you have written a lot of badh test code already but maybe doing this in e2e would be better?
All tests have a custom --roor/--runroot by default including remote. Also I assume these checks are expensive and slow? In this case it would be another reason for e2e as they run parallel so slow tests are not that bad there.

Reply:
Poring over actual data, they can be time-intensive.  The ones we set up for testing are small enough that the new tests finish in about 15 seconds on my dev machine.

Reply:
To be clear I don't want you to force you to write it all to e2e if you don't want this.
15s is fine I guess, I am working on strategies to make the system tests faster so I will keep this in mind but I know there are worse offenders so not a reason to block this one